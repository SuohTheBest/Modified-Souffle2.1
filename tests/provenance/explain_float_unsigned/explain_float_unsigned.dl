// Souffle - A Datalog Compiler
// Copyright (c) 2020, The Souffle Developers. All rights reserved
// Licensed under the Universal Permissive License v 1.0 as shown at:
// - https://opensource.org/licenses/UPL
// - <souffle root>/licenses/SOUFFLE-UPL.txt

//
// explain_float_unsigned.dl
// This code tests the provenance explain interface for a simple path example.
//

.pragma "provenance" "explain"

.decl Float(x: float)
Float(1.1).

.decl FtoI(x: number)
FtoI(to_number(x)) :- Float(x).

.decl VUnsigned(x:unsigned)
VUnsigned(0).
VUnsigned(1).
VUnsigned(2).
VUnsigned(3).
VUnsigned(5).


.decl edgeUnsigned(x: unsigned, y: unsigned)
edgeUnsigned(x, y) :- VUnsigned(x), VUnsigned(y), x + 1 = y.
edgeUnsigned(x, y) :- edgeUnsigned(y, x).

.decl pathUnsigned(x: unsigned, y: unsigned)
pathUnsigned(x, y) :- edgeUnsigned(x, y).
pathUnsigned(x, z) :- edgeUnsigned(x, y), pathUnsigned(y, z).



.output FtoI, pathUnsigned
