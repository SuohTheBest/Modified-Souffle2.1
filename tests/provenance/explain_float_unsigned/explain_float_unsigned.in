explain FtoI(1)
explain pathUnsigned(0,3)
explainnegation pathUnsigned(0,5)
2
2
exit