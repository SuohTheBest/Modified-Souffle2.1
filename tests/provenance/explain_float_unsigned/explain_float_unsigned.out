Float(1.100000) 
------------(R1)
    FtoI(1)     
                                 VUnsigned(1) VUnsigned(2) 2 = 2  subproof edgeUnsigned(0)   
                                 ----------------------------(R1) ---------------------(R1)  
VUnsigned(0) VUnsigned(1) 1 = 1         edgeUnsigned(1, 2)           pathUnsigned(2, 3)      
----------------------------(R1) -------------------------------------------------------(R2) 
       edgeUnsigned(0, 1)                            pathUnsigned(1, 3)                      
-----------------------------------------------------------------------------------------(R2)
                                     pathUnsigned(0, 3)                                      
edgeUnsigned(0, 2) x pathUnsigned(2, 5) x 
--------------------------------------(R2)
            pathUnsigned(0,5)             
